src/pages/UserSettings/UserSettings.tsx

```diff
@@ -0,0 +1,209 @@
+import * as EmailValidator from "email-validator";
+import { useEffect, useRef, useState } from "react";
+import { useMutation } from "react-query";
+import Button from "../../components/Button";
+import Spinner from "../../components/Spinner";
+import TextField from "../../components/TextField";
+import Toggle from "../../components/Toggle";
+import Column from "../../components/layout/Column";
+import Row from "../../components/layout/Row";
+import ConfirmationModal from "../../components/modals/ConfirmationModal";
+import { Constants } from "../../constants";
+import { useApi } from "../../contexts/ApiProvider";
+import { useUser } from "../../contexts/UserProvider";
+import { SettingShort } from "../../data/models/Setting";
+export default function UserSettings() {
+  const [toggled, setToggled] = useState(false);
+  const { user, deleteUser } = useUser();
+  const { api } = useApi();
+  const [hasEmailAlerts, setHasEmailAlerts] = useState(
+    user!.settings.find((s) => s.key === "notify" && s.enabled === true) !==
+      undefined,
+  );
+  // we only care about the time of this date
+  const [emailTime, setEmailTime] = useState(
+    user!.settings.find((s) => s.key === "notify")?.value ?? "09:00",
+  );
+  const [email, setEmail] = useState("");
+  const [changeEmailState, setChangeEmailState] = useState("");
+
+  const [username, setUsername] = useState("");
+  const [changeUsernameState, setChangeUsernameState] = useState("");
+
+  const [emailTimeNetworkingState, setEmailTimeNetworkingState] = useState(
+    Constants.FINISHED_MUTATION_TEXT,
+  );
+  // For email updating based on timeout
+  const updateEmailTimeout = useRef<number | undefined>();
+  const updateEmailHandler = () => {
+    updateEmailAlerts.mutate(hasEmailAlerts);
+  };
+
+  // update time when time is updated
+  const [isFirstRender, setIsFirstRender] = useState(false);
+  useEffect(() => {
+    if (isFirstRender) {
+      setIsFirstRender(false);
+    } else {
+      clearTimeout(updateEmailTimeout.current);
+      updateEmailTimeout.current = setTimeout(updateEmailHandler, 500);
+    }
+  }, [emailTime]);
+
+  // DANGEROUS delete account mutation
+  const deleteAccountMutation = useMutation({
+    mutationFn: () => api.delete("/user/"),
+    onSuccess: () => {
+      deleteUser();
+    },
+  });
+
+  // Updating email alerts setting
+  const updateEmailAlerts = useMutation({
+    // Only way to have optional parameters for mutationFn
+    // https://github.com/TanStack/query/issues/4264#issuecomment-1268054812
+    mutationFn: (enabled: boolean) =>
+      api
+        .post<SettingShort>("/user/settings/", {
+          key: "notify",
+          value: emailTime,
```

Comment:
does this incorporate the timezone? it seems like this just sends the time (ex. 15:30) but I think the backend code expects a timezone? https://github.com/jonathanjma/HappinessApp/blob/9605b958649c50178cfeffacc0c1e03b10ceafa8/happiness-backend/jobs/jobs.py#L146

Reply:
good point, I will add this.